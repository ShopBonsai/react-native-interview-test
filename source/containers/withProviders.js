import { compose } from "recompose"

import reduxStore from "../store"

import withRedux from "./withRedux"
import withSafeArea from "./withSafeArea"

export default compose(withRedux(reduxStore), withSafeArea)
