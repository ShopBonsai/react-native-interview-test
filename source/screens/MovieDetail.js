import React, { Component } from "react"
import PropTypes from "prop-types"
import { View, Image, Text, ScrollView } from "react-native"
import { connect } from "react-redux"

import { Button } from "../atoms"
import MovieDetailItem from "../molecules/MovieDetailItem"
import Dimensions from "../configs/Dimensions"
import * as actions from "../redux/actions"

const style = {
  scrollContainer: {
    flex: 1,
  },
  container: {
    flex: 1,
    alignItems: "flex-start",
    justifyContent: "flex-start",
  },
  moviePosterStyle: {
    width: Dimensions.screenWidth,
    height: Dimensions.screenHeight / 2,
  },
  detailContainer: {
    margin: 10,
  },
  movieTitle: {
    fontSize: 24,
    fontWeight: "bold",
  },
  detailLastRow: {
    flex: 1,
    flexDirection: "row",
    justifyContent: "space-between",
  },
  movieDetailBottons: {
    flex: 1,
  },
}

class MovieList extends Component {
  onFavouritePress = () => {
    const { addToFavourites, movieId } = this.props
    addToFavourites(movieId)
  }
  render() {
    const { favourites, movieId, movies } = this.props
    const selectedMovie = movies.find(movie => movie.id === movieId)
    const isFavourite = favourites[movieId]
    if (selectedMovie) {
      const { date, genre, image, inventory, price, title } = selectedMovie
      return (
        <View style={style.container}>
          <Image style={style.moviePosterStyle} source={image} />
          <ScrollView style={style.detailContainer}>
            <Text style={style.movieTitle}>{title}</Text>
            <MovieDetailItem label="Show Time:" value={date} />
            <MovieDetailItem label="Genre:" value={genre.join(" | ")} />
            <MovieDetailItem label="Tickets Left:" value={inventory} />
            <MovieDetailItem label="Ticket Price:" value={`$${price}`} />
            <View style={style.detailLastRow}>
              <Button style={style.movieDetailBottons} onPress={this.onFavouritePress}>
                {isFavourite? "Unlike" :"Like"}
              </Button>
            </View>
          </ScrollView>
        </View>
      )
    } else {
      return (
        <View style={{ flex: 1 }}>
          <Text>Movie Not Found</Text>
        </View>
      )
    }
  }
}
MovieList.propTypes = {
  addToFavourites: PropTypes.func,
  favourites: PropTypes.objectOf(PropTypes.number),
  movieId: PropTypes.string,
  movies: PropTypes.arrayOf(PropTypes.object),
}
const mapStateToProps = state => {
  return {
    movies: state.movies.movies,
    favourites: state.favourites.favourites,
  }
}
const mapDispatchToProps = dispatch => ({
  fetchMovies: (skip, limit) => {
    dispatch(actions.fetchMovies(skip, limit))
  },
  addToFavourites: id => {
    dispatch(actions.addToFavourites(id))
  },
})

export default connect(mapStateToProps, mapDispatchToProps)(MovieList)
