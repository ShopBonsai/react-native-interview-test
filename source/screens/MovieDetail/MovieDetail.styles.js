import styled from 'styled-components'
import { ScrollView, Text } from 'react-native'
import FastImage from 'react-native-fast-image'
import themes from '../../themes'

export const MovieDetailContainer = styled(ScrollView)`
  flex: 1;
  background-color: ${themes.theme.colors.backgroundColor.secondary};
`
export const MovieDetailContent = styled(ScrollView)`
  flex: 1;
  flex-direction: column;
`
export const MovieDetailCover = styled(FastImage)`
  flex: 1;
  height: 180px;
`
export const MovieDetailTitle = styled(Text)`
  color: white;
  font-size: 25px;
  padding: 10px 20px;
`

export const MovieDetailPrice = styled(Text)`
  color: white;
  font-size: 18px;
  padding: 10px 20px;
`
