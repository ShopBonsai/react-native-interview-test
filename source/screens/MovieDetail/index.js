import React from 'react'
import { MovieDetailContainer, MovieDetailContent } from './MovieDetail.styles';

export default function MovieDetail(props) {
  return (
    <MovieDetailContainer>
      <MovieDetailContent>

      </MovieDetailContent>
    </MovieDetailContainer>
  )
}