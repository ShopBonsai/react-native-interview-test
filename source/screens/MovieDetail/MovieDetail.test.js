// libraries
import React from 'react'
import renderer from 'react-test-renderer'
import MovieDetail from './index'

// components
import withProviders from '../../containers/withProviders'

it('Renders MovieDetail correctly', () => {
  const Component = withProviders(MovieDetail)
  const tree = renderer
    .create(<Component />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
