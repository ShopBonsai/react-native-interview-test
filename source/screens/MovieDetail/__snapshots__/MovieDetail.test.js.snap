// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders MovieDetail correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "black",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <RCTScrollView
    className="sc-bdVaJa lcApuY"
  >
    <View>
      <RCTScrollView
        className="sc-bwzfXH bpwjAU"
      >
        <View>
          <View
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                undefined,
              ]
            }
          >
            <FastImageView
              className="sc-htpNat gbVinx"
              resizeMode="cover"
              source={
                Object {
                  "cache": "immutable",
                  "uri": null,
                }
              }
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            />
          </View>
          <Text
            className="sc-bxivhb hWLKnT"
          />
          <Text
            className="sc-ifAKCX eWIMju"
          >
            Price: U$
          </Text>
          <Text
            className="sc-ifAKCX eWIMju"
          >
            Genre:
          </Text>
        </View>
      </RCTScrollView>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
